`--target ESNext` does not include lib.dom.d.ts by default

**TypeScript Version:**
2.2.1

**Code**

Try running tsc in a file that contains the following (`tsc test.ts` for example):
```ts
var a  = new Map();
```

**Expected behavior:**
Not to get an error message in the output (`test.js` gets generated though)

**Actual behavior:**

The output of the command above (`tsc test.ts`) is as follows:
> test.ts(1,15): error TS2693: 'Map' only refers to a type, but is being used as a value here.

`tsc .\test.js --target esnext` gives the following error:
> node_modules/@types/highlight.js/index.d.ts(19,40): error TS2304: Cannot find name 'Node'.

Guess that something related with my project

`tsc .\test.js --target es6` doesn't output any error/warning

`tsc .\test.js` with the tsconfig.json and target `es6` next outputs:
> test.ts(1,15): error TS2693: 'Map' only refers to a type, but is being used as a value here.

@blakeembrey specifying `test.ts` in the `tsconfig.json` and with `es6` and `esnext` returns:
> node_modules/@types/highlight.js/index.d.ts(19,40): error TS2304: Cannot find name 'Node'.

even if I have excluded `node_modules`:

```json
{
    "compilerOptions": {
        "target": "esnext"
    },
    "files": [
        "test.ts"
    ],
   "exclude": [
        "node_modules"
    ]
}
```
Not sure why I'm getting that message but I guess the output is the expected then?

--- a/Gulpfile.ts
+++ b/Gulpfile.ts
@@ -162,7 +162,10 @@ const librarySourceMap = [
 
     // JavaScript + all host library
     { target: "lib.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(hostsLibrarySources) },
-    { target: "lib.es6.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") }
+    { target: "lib.es6.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") },
+    { target: "lib.es2016.full.d.ts", sources: ["header.d.ts", "es2016.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") },
+    { target: "lib.es2017.full.d.ts", sources: ["header.d.ts", "es2017.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") },
+    { target: "lib.esnext.full.d.ts", sources: ["header.d.ts", "esnext.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") },
 ].concat(es2015LibrarySourceMap, es2016LibrarySourceMap, es2017LibrarySourceMap, esnextLibrarySourceMap);
 
 const libraryTargets = librarySourceMap.map(function(f) {

--- a/Jakefile.js
+++ b/Jakefile.js
@@ -205,7 +205,10 @@ var librarySourceMap = [
 
     // JavaScript + all host library
     { target: "lib.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(hostsLibrarySources) },
-    { target: "lib.es6.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") }
+    { target: "lib.es6.d.ts", sources: ["header.d.ts", "es5.d.ts"].concat(es2015LibrarySources, hostsLibrarySources, "dom.iterable.d.ts") },
+    { target: "lib.es2016.full.d.ts", sources: ["header.d.ts", "es2016.d.ts"].concat(hostsLibrarySources, "dom.iterable.d.ts") },
+    { target: "lib.es2017.full.d.ts", sources: ["header.d.ts", "es2017.d.ts"].concat(hostsLibrarySources, "dom.iterable.d.ts") },
+    { target: "lib.esnext.full.d.ts", sources: ["header.d.ts", "esnext.d.ts"].concat(hostsLibrarySources, "dom.iterable.d.ts") },
 ].concat(es2015LibrarySourceMap, es2016LibrarySourceMap, es2017LibrarySourceMap, esnextLibrarySourceMap);
 
 var libraryTargets = librarySourceMap.map(function (f) {

--- a/src/compiler/utilities.ts
+++ b/src/compiler/utilities.ts
@@ -4202,12 +4202,13 @@ namespace ts {
     export function getDefaultLibFileName(options: CompilerOptions): string {
         switch (options.target) {
             case ScriptTarget.ESNext:
+                return "lib.esnext.full.d.ts";
             case ScriptTarget.ES2017:
-                return "lib.es2017.d.ts";
+                return "lib.es2017.full.d.ts";
             case ScriptTarget.ES2016:
-                return "lib.es2016.d.ts";
+                return "lib.es2016.full.d.ts";
             case ScriptTarget.ES2015:
-                return "lib.es6.d.ts";
+                return "lib.es6.d.ts";  // We don't use lib.es2015.full.d.ts due to breaking change.
             default:
                 return "lib.d.ts";
         }

